UI/src/view/csvReportsWorkList.js

```diff
@@ -0,0 +1,489 @@
+(function (root) {
+  root.csvReportsWorkList = function () {
+    WorkListView.call(this);
+    var me = this;
+    this.hrefDir = "#work-list/csvReports";
+    this.title = 'Raportointityökalu';
+    var backend;
+    var municipalities;
+    var assetsList;
+    var refresh;
+
+
+    this.initialize = function (mapBackend) {
+      backend = mapBackend;
+      me.bindEvents();
+    };
+
+
+    this.bindEvents = function () {
+      eventbus.on('csvReports:select', function (listP) {
+        $('.container').hide();
+        $('#work-list').show();
+        $('body').addClass('scrollable');
+        me.generateWorkList(listP);
+      });
+
+    };
+
+    function getMunicipalities() {
+      if (_.isEmpty($("#municipalities_search").find('option'))) {
+        if (_.isEmpty(municipalities)) {
+
+          backend.getMunicipalities().then(function (result) {
+            municipalities = result;
+            setMunicipalities();
+          });
+
+        } else
+          setMunicipalities();
+      }
+    }
+
+    function getAssetsType() {
+      if (_.isEmpty($("#assets_search").find('option'))) {
+        if (_.isEmpty(assetsList)) {
+
+          backend.getAssetTypes().then(function (result) {
+            assetsList = result;
+            setAssets();
+          });
+        } else
+          setAssets();
+      }
+    }
+
+    function setMunicipalities() {
+      $('#municipalities_search').append($('<option>', {
+        value: "1000",
+        text: "Kaikki kunnat"  /* all municipalities */
+      }));
+
+      _.forEach(municipalities, function (municipality) {
+        $('#municipalities_search').append($('<option>', {
+          value: municipality.id,
+          text: municipality.name
+        }));
+      });
+    }
```

Comment:
I think maybe it should be better like this:

```javascript
    function setMunicipalities() {
      var allMunicipalitiesChoice = {id: "1000", name: "Kaikki kunnat"};
      var allMunicipalities = _.concat(municipalities, allMunicipalitiesChoice);

      _.forEach(allMunicipalities , function (municipality) {
        $('#municipalities_search').append($('<option>', {
          value: municipality.id,
          text: municipality.name
        }));
      });
    }
```

Reply:
applied